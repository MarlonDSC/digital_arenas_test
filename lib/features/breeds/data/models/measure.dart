import 'package:json_annotation/json_annotation.dart';

part 'measure.g.dart';

@JsonSerializable()
class Measure {
  final String imperial;
  final String metric;

  Measure({required this.imperial, required this.metric});

  factory Measure.fromJson(Map<String, dynamic> json) => _$MeasureFromJson(json);
  Map<String, dynamic> toJson() => _$MeasureToJson(this);
}

