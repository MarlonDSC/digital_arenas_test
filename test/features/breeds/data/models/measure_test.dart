import 'package:flutter_test/flutter_test.dart';
import 'package:approval_tests/approval_tests.dart';
import 'package:inmo_mobile/features/breeds/data/models/measure.dart';

void main() {
  group('Measure', () {
    test('verify basic serialization', () {
      final measure = Measure(
        imperial: '50-75 lbs',
        metric: '23-34 kg',
      );

      Approvals.verifyAsJson(
        measure,
        options: const Options(
          reporter: DiffReporter(),
        ),
      );
    });

    test('verify complex measurement serialization', () {
      final measure = Measure(
        imperial: '22-25 inches',
        metric: '56-64 cm',
      );

      Approvals.verifyAsJson(
        measure,
        options: const Options(
          reporter: DiffReporter(),
        ),
      );
    });

    test('verify JSON roundtrip', () {
      final originalMeasure = Measure(
        imperial: '60-80 lbs',
        metric: '27-36 kg',
      );

      final json = originalMeasure.toJson();
      final deserializedMeasure = Measure.fromJson(json);

      Approvals.verifyAsJson(
        {
          'original': originalMeasure,
          'serialized': json,
          'deserialized': deserializedMeasure,
        },
        options: const Options(
          reporter: DiffReporter(),
        ),
      );
    });

    test('verify fromJson factory with sample payload', () {
      const json = {
        'imperial': '20-25 inches',
        'metric': '51-64 cm',
      };

      final measure = Measure.fromJson(json);

      Approvals.verifyAsJson(
        {
          'input': json,
          'parsed': measure,
        },
        options: const Options(
          reporter: DiffReporter(),
        ),
      );
    });
  });
}