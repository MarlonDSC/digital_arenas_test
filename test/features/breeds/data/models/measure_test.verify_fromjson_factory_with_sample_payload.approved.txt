# This file was generated by approval_tests. Please do not edit.

{
  "input": {
    "imperial": "20-25 inches",
    "metric": "51-64 cm"
  },
  "parsed": {
    "imperial": "20-25 inches",
    "metric": "51-64 cm"
  }
}